src/main/java/ch/ricardo/attributeextractor/model/Language.java

```
@@ -0,0 +1,45 @@
+package ch.ricardo.attributeextractor.model;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonValue;
+import java.util.Set;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+import lombok.AllArgsConstructor;
+
+@AllArgsConstructor
+public enum Language {
+  DE("de"),
+  FR("fr");
+
+  private String label;
+
+  @Override
+  @JsonValue
+  public String toString() {
+    return label;
+  }
+
+  @JsonCreator
+  public static Language fromString(String label) {
+    if (label == null) {
+      return null;
+    }
+    for (Language language : values()) {
+      if (language.label.equals(label)) {
```

Not a blocker but maybe equalsIgnoreCase ?

---

It depends of how strict we want to be. Do we want to accept `Accept-Language:DE` as a header, or `lang=DE`, or `{ "DE": "Möbel" }`? IMO it's better to be strict and fail fast (this would fail on validation of the initial request).
Btw, I took this class from the `article-classification-domain` module (and I'd say it should be in some common module).